site/content/Architecture/_index.md

```diff
@@ -1,15 +1,232 @@
 ---
 title: "Architecture"
 weight: 4
+mermaid: true
 description: >
     How is VHS implemented internally?
 ---
 
 ## Introduction
-`vhs` is designed for flexibility and operates on the concept of a data flow that originates with a [source](/vhs/reference/#sources)
-and terminates with one or more [sinks](/vhs/reference/#sinks). Sources may capture network data, read data from files, etc. Sinks may
-write data to cloud or local storage, standard output, or send data to other destinations. Along the way, data may pass
-through a series of input [modifiers](/vhs/reference/#input-modifiers) and [formats](/vhs/reference/#input-formats) and output
-[modifiers](/vhs/reference/#output-modifiers) and [formats](/vhs/reference/#output-formats) that transform the data.
 
-More architectural details coming soon.
\ No newline at end of file
+`vhs` is a network traffic utility that works by chaining modules or plugins. It offers a high-performance concurrent
```

Comment:
Maybe:

> "chaining components loaded from plugins"

Reply:
Good suggestion, incorporated in 7ecc5b3cf1b695460fe76db6347fbb5ad3136c06.